src/charm.py

```diff
@@ -365,16 +377,15 @@ def _handle_subscription_state(self):
             update_configuration(contract_url)
             self._state.contract_url = contract_url
 
-        try:
-            status = get_status_output(self.ssl_env)
-        except ProcessExecutionError as e:
-            self.unit.status = BlockedStatus(str(e))
-            return
-        if status["attached"] and (config_changed or token_changed):
-            detach_subscription(self.ssl_env)
-            self._state.hashed_token = None
-
         if not token:
+            try:
+                status = get_status_output(self.ssl_env)
+            except ProcessExecutionError as e:
+                self.unit.status = BlockedStatus(str(e))
+                return
+            if status["attached"] and (config_changed or token_changed):
+                detach_subscription(self.ssl_env)
+                self._state.hashed_token = None
             self.unit.status = BlockedStatus("No token configured")
             return
         elif config_changed or token_changed:
```

Comment:
The logger call below arguably belongs in the function itself, not here.

Reply:
Sure! I can include that as an extra refactor commit.